syft/core/frameworks/torch/hook.py

```diff
@@ -14,23 +14,21 @@
 class TorchHook(object):
 
     def __init__(self, local_worker=None, is_client=True, verbose=True, queue_size=0):
-
         self.local_worker = local_worker
+
+        if not hasattr(torch, 'torch_hooked'):
+            torch.torch_hooked = 0
+        else:
+            torch.torch_hooked += 1
+
         if (self.local_worker is None):
 
             # Every TorchHook instance should have a local worker which is responsible for
             # interfacing with other workers. The worker interface is what allows the Torch
             # specific code in TorchHook to be agnostic to the means by which workers communicate
             # (such as peer-to-peer, sockets, through local ports, or all within the same process)
 
-            if (hasattr(torch, 'local_worker')):
-                self.local_worker = torch.local_worker
-                if (verbose):
-                    print("Torch seems to already have a local_worker object... \
-                          using that one instead...")
-            else:
-                self.local_worker = workers.VirtualWorker(
-                    hook=self, is_client_worker=is_client, queue_size=queue_size)
+            self.local_worker = workers.VirtualWorker(hook=self, is_client_worker=is_client, queue_size=queue_size)
```

Comment:
what was the thought behind removing this part?

Reply:
It's just that as for now the local worker is declared to be an attribute of hook and not of torch anymore, so ` -            if (hasattr(torch, 'local_worker')): ` will never be True.
Therefore, I added another torch attribute to keep track of instanciation of torchhook. Also, I tried to see in what extend we could instanciate several instances of torchhook (what parts should run several times, or only once), for us to run all the tests at once, but it revealed tricky so I dropped this topic. Nvm, could be an issue or a todo